src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -449,8 +449,8 @@ Status MutableObjectManager::SetErrorInternal(const ObjectID &object_id) {
 }
 
 Status MutableObjectManager::SetErrorAll() {
-  absl::ReaderMutexLock guard(&destructor_lock_);
-
+  absl::ReaderMutexLock destructor_guard(&destructor_lock_);
+  absl::ReaderMutexLock channel_guard(&channel_lock_);
```

Comment:
In this change, SetErrorAll holds reader lock to channel_lock_. Inside, it calls SetErrorInternal which calls GetChannel, which in turn acquires reader lock to channel_lock_. This is a recursive lock and it not something we want.

Can we change SetErrorInternal to accept (object_id, Channel*), and here we pass in the channel*, since we are already iterating the map?

to prevent this, I recommend adding ABSL_LOCKS_EXCLUDED to GetChannel.

Reply:
also, no outside code uses GetChannel, so let's move it into private, and add `const`.

Reply:
fixed, SetErrorInternal taking channel as well now, and added absl_locks_excluded to GetChannel

made GetChannel private, but can't make it const because it needs to return a non-const ref to channel